geometry/diamond_box.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "geometry/bounding_box.hpp"
+#include "geometry/point2d.hpp"
+
+namespace m2
+{
+// Bounding box for a set of points on the plane, rotated by 45
+// degrees.
```

Comment:
Not really: if you take the bounding box and rotate it, the resulting rectangle may differ from the DiamondBox. Besides, there may be bounding boxes different from "the" bounding box.

The smallest rectangle containing |points| whose sides are parallel to the axes rotated by 45 degrees.
or
The smallest rectangle containing |points| whose sides form 45-degree angles with coordinate axes.

Reply:
From my point of view in the common sense the term "Bounding Box" means exactly what BoundingBox class does, i.e. the smallest rectangle containing all points with sides parallel to the axes, therefore my comment does not contradict this definition.

Reply:
May be; yet the comment sounds ambiguous to me.
The code is simple enough, so it is probably not a problem. Up to you.